src/main/java/io/kontur/disasterninja/service/UserProfileService.java

```diff
@@ -42,4 +46,21 @@ public UserDto updateUser(UserDto userDto) {
             throw new WebApplicationException("Can't update current user", HttpStatus.BAD_GATEWAY);
         }
     }
+
+    /**
+     * Checks if user has access to any of the provided features
+     * @param featureNames feature names to check user access
+     * @return true if user has access to at least one of the provided features,
+     *         false otherwise
+     */
+    public boolean userHasAccessToFeature(List<String> featureNames) {
+        List<AppSummaryDto> appSummaries = client.getAppsList();
+        for (AppSummaryDto appSummary : appSummaries) {
+            AppDto app = client.getApp(appSummary.getId());
+            if (app.getFeatures().stream().anyMatch(feature -> featureNames.contains(feature.getName()))) {
```

Comment:
Why don't we put everything in a single stream computation like
```java
client.getAppsList().stream().flatMap(summary -> client.getApp(summary.getId()).getFeatures().stream()).anyMatch(feature -> featureNames.contains(feature.getName()))
```
or something along those lines? 😅

Reply:
I am removing authorization at all for now. We need to investigate this more. Checking if feature is available for a user in any app, is too time consuming. We need to add appId parameter or think of a new way to check access.